Add ability to refetch permissions on demand (#8955)
marmelab/react-admin: docs/usePermissions.md

}
```

## Refreshing permissions

> Permissions are loaded on page load and cached. If your application requires permissions to be refreshed, for example after a change modifying user permissions, you can use `refetch` function to trigger reload.

Reviewer: ```suggestion
Permissions are loaded when the app loads and then cached. If your application requires permissions to be refreshed, for example after a change modifying user permissions, you can use `refetch` function to trigger reload.
```

```jsx
const GrantAdminPermissionsButton = () => {
    const record = useRecordContext();
    const [ update ] = useUpdate();
    const { refetch } = usePermissions();

    const handleClick = () => {

>         update(
>             "users",
>             { id: record.id, data: { admin: true }, previousData: record },
>         );
>         refetch();

Reviewer: ```suggestion
        update(
            "users",
            { id: record.id, data: { admin: true }, previousData: record },
            { onSuccess: refetch },
        );
```

    }

    return (
        <Button onClick={handleClick}>
            Make user an admin
        </Button>
    )
}

> ```

Reviewer: Also, this whole code block should be wrapped with `{% raw %}` `{% endraw %}` to avoid issues with jekyll. Can you wrap it?
